plugins/uefi/fu-uefi-device.c

```diff
@@ -424,6 +424,8 @@ fu_uefi_device_write_firmware (FuDevice *device, GBytes *fw, GError **error)
 	/* update the firmware before the bootloader runs */
 	if (fu_device_get_metadata_boolean (device, "RequireShimForSecureBoot"))
 		flags |= FU_UEFI_BOOTMGR_FLAG_USE_SHIM_FOR_SB;
```

Comment:
Just a side note I noticed while reviewing this.  `FU_UEFI_BOOTMGR_FLAG_USE_SHIM_FOR_SB` doesn't really do justice to how this actually works.  If you have shim installed but don't have `RequireShimForSecureBoot` set, it's going to still use shim as I can tell (since it's just looking for the file existing).  It's probably a major corner case though, most people unsetting `RequireShimForSecureBoot` probably won't have shim installed in the first place anyway.

Reply:
> most people unsetting RequireShimForSecureBoot probably won't have shim installed in the first place anyway

Right; I think it was designed as an _escape hatch_ for those people, rather than a way to disable it.